### PR TITLE
Export application name and version in tests

### LINKED ISSUE #15013: Application name and version are not available in test
As discuss here: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Application.20name.20and.20version, 
`quarkus.application.name` and `quarkus.application.version` are not available in test mode. 

To be consistent, those configuration should always be available and fallback to artifact id/version if not set.

### PR DESCRIPTION
**Pull request by glefloch**
This export `quarkus.application.name` and `quarkus.application.version` properties. This used to resolved model to extract data in the boostrap phase. 

close #15013

### CODE REVIEW

**Comment by glefloch** [comment]
@aloubyansky could you have a look? I will fix the build, but I would like to be sure I'm heading in the right direction.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java [hunk 1]
```diff
@@ -159,7 +159,15 @@ public CuratedApplication bootstrap() throws BootstrapException {
                 appModelFactory.setEnableClasspathCache(true);
             }
         }
-        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig);
+        CurationResult curationResult = appModelFactory.resolveAppModel();
+
+        if (mode == Mode.TEST || test) {
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java:164]: We could probably do this in any mode.

**Comment by aloubyansky** [comment]
I think it is, thanks @glefloch

**Comment by glefloch** [comment]
@cescoffier the kafka test keeps failing due to a serialization issue after the devmode reload, I don't see how the failure can be related to my branch, do you have an idea?

**Comment by cescoffier** [comment]
Hard to say, but having two failures is highly suspicious.

**Comment by cescoffier** [comment]
Oh yeah! The kafka connector uses the application name as consumer group is.

**Comment by glefloch** [comment]
Setting it to an empty value fix the test

**Comment by glefloch** [comment]
Finally got a green CI :), @aloubyansky @cescoffier could you review it ?

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java [hunk 2]
```diff
@@ -159,7 +159,20 @@ public CuratedApplication bootstrap() throws BootstrapException {
                 appModelFactory.setEnableClasspathCache(true);
             }
         }
-        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig);
+        CurationResult curationResult = appModelFactory.resolveAppModel();
+        if (curationResult.getApplicationModel().getApplicationModule() != null
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java:163]: `curationResult.getApplicationModel().getAppArtifact()` would be simpler and a bit more reliable here.

**Comment by aloubyansky** [comment]
@phillip-kruger could you please check the Open API test changes? Should be ok but just in case.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java (hunk 2 above)
Comment (glefloch) [reply independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java]: right, I pushed a fix.

**Review by aloubyansky — APPROVED** [review]

**Comment by glefloch** [comment]
@phillip-kruger could you have a look at Open api test change to make sure I haven't introduced any regression ?

**Review by phillip-kruger — APPROVED** [review]
Lgtm

**Comment by glefloch** [comment]
No problem, thanks for review. 
@aloubyansky can we merge this one? the gradle failure does not seem to be related.

**Review by aloubyansky — APPROVED** [review]

**Comment by aloubyansky** [comment]
Yes, thanks @glefloch

**Pull request merged**